File: composition.py
class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age

# 継承
class Student(Person):
    def __init__(self, name, age, grade):
        super().__init__(name, age)
        self.grade = grade

# インスタンス化
yamada = Student("山田", 20, 2)

# コンポジション
class Student_:
    def __init__(self, person, grade):
        self.name = person.name
        self.age = person.age
        self.grade = grade

# インスタンス化
yamada = Student(Person("山田", 20), 2)

File: ts/composition.ts
class Person {
  name: string
  age: number
  constructor(name: string, age: number) {
    this.name = name
    this.age = age
  }

  public sayHello = () => {
    console.log(`Hello, I'm ${this.name} . I ${this.age} years old.`)
  }
}

const person = new Person('mahito', 23)
person.sayHello()

class Student {
  name: string
  age: number
  belong: string
  constructor(person: Person, belong: string) {
    this.name = person.name
    this.age = person.age
    this.belong = belong
  }

  public sayHello = () => {
    console.log(`Hello, I'm ${this.name} . I ${this.age} years old and belong to ${this.belong}`)
  }
}

const student = new Student(person, 'Oita university')
student.sayHello()